preview/views/ViewBitmap.hpp

```diff
@@ -10,16 +10,16 @@ namespace services
         : public View
     {
     public:
-        explicit ViewBitmap(infra::Bitmap& bitmap);
+        explicit ViewBitmap(const infra::Bitmap& bitmap);
 
         // Implementation of View
         void Paint(hal::Canvas& canvas, infra::Region boundingRegion) override;
 
-        void Source(infra::Bitmap& source);
-        infra::Bitmap& Source() const;
+        void Source(const infra::Bitmap& source);
+        const infra::Bitmap& Source() const;
 
     private:
-        infra::Bitmap& source;
+        infra::Bitmap source;
```

Comment:
Because you are copying an object with MemoryRange, when the range is somehow invalidated externally, wouldn't this object be out of sync ? Could this cause an unintentional behavior change?

Reply:
The bitmap indeed lives outside of the view, but I don't think there is an expectation that a view holds the storage